lib/ueberauth/strategy/oidc.ex

```diff
@@ -14,7 +14,9 @@ defmodule Ueberauth.Strategy.OIDC do
   """
   def handle_request!(conn) do
     provider_id = conn |> get_options!() |> get_provider()
+
     params = params_from_conn(conn)
+    params = with_state_param([], conn) |> Enum.into(params)
```

Comment:
Could use the existing `params_from_conn()` for this, and directly get the `conn.private[:ueberauth_state_param]
` over there and add it to the returned map.

We need to add a test to check if the state is correctly retrieved too.

Reply:
Yep. I changed it.